idea/testData/codeInsight/surroundWith/if/moveDeclarationsOut/order/valAndClass.kt.after

```diff
@@ -3,9 +3,9 @@ fun foo() {
         fun test() {}
     }
 
-    val d: A
+    val d =
     if (<caret>) {
-        d = A()
+        A()
```

Comment:
I think this is quite hard to fix...it's problem with locating `dummyFirstStatement` and inserting generated statement. I'm not even sure that it is possible to fix this.